runtime/vm/JFRConstantPoolTypes.hpp

```diff
@@ -413,6 +415,110 @@ class VM_JFRConstantPoolTypes {
 		return;
 	}
 
+	const char *getSystemProp(const char *propName) {
+		J9VMSystemProperty *jvmInfoProperty = NULL;
+		J9InternalVMFunctions *vmFuncs = _vm->internalVMFunctions;
+		const char *value = "";
+		UDATA getPropertyResult = vmFuncs->getSystemProperty(_vm, propName, &jvmInfoProperty);
+		if (J9SYSPROP_ERROR_NOT_FOUND != getPropertyResult) {
+			value = jvmInfoProperty->value;
+		}
+		return value;
+	}
+
+    void initJVMInfo() {
+		const char *javaVMName = getSystemProp("java.vm.name");
```

Comment:
you should put the basic logic in a function since its repeated a few times.

Reply:
Could you elaborate what basic logic are you referring to?

Reply:
Taking the value of a sys prop, allocating a j9utf8 and placing the sysprop into the j9utf8. Basically, if you are doing something repeatedly, you should put it in a function.

Reply:
I've refactored out the string building part.